backend/debug/is_parameter.mli

```diff
@@ -0,0 +1,29 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*                  Mark Shinwell, Jane Street Europe                     *)
+(*                                                                        *)
+(*   Copyright 2014--2018 Jane Street Group LLC                           *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+[@@@ocaml.warning "+a-4-30-40-41-42"]
+
+(** Whether a variable is a local or a function parameter. *)
+
+type t = private
+  | Local
+  | Parameter of { index : int }
+
+val local : t
+
+val parameter : index:int -> t
+
+val join : t -> t -> t
```

Comment:
What is the use case (or semantics) for this function ? I'm concerned that the implementation errors on distinct parameters without an argument saying why it shouldn't happen.

Reply:
Not used at the moment, so deleted.